spec/requests/restaurants_spec.rb

```diff
@@ -0,0 +1,132 @@
+require 'rails_helper'
+
+RSpec.describe 'Restaurants', type: :request do
+  before(:all) do
+    FactoryBot.create_list(:restaurant, 10)
+  end
+
+  describe 'GET /restaurants' do
+    before do
+      get restaurants_path
+    end
+
+    it 'return https success' do
+      expect(response).to have_http_status(200)
+    end
+
+    it 'has the right objects count' do
+      arr = response_body
+      expect(arr.count).to eq(Restaurant.all.count)
+    end
+
+    it 'contains mandatory attributes' do
+      arr = response_body
+      arr.each do |json_rest|
+        expect(json_rest.keys).to include('id', 'name', 'cuisine', 'accept10bis')
+      end
+    end
+  end
+
+  describe 'GET /restaurant/:id' do
+    subject { Restaurant.all.sample }
+    it 'return https success' do
+      get restaurant_path(subject.id)
+      expect(response).to have_http_status(200)
+    end
+
+    it 'returns :not_found status on invalid uuid' do
+      get restaurant_path('this-is-not-a-uuid')
+      expect(response).to have_http_status(:not_found)
+    end
+
+    it 'returns the correct record' do
+      get restaurant_path(subject.id)
+      restaurant_matcher(subject, response_body)
+    end
+  end
+
+  describe 'POST /restaurants' do
+    it 'returns http success' do
+      post '/restaurants', params: FactoryBot.attributes_for(:restaurant)
```

Comment:
if you use `FactoryBot.build(:restaurant)`, will it work?

Reply:
nop, 
it'll work with FactoryBot.build(:restaurant).attributes